redash/query_runner/dgraph.py

```diff
@@ -0,0 +1,127 @@
+import json
+import re
+
+try:
+    import pydgraph
+    enabled = True
+except ImportError:
+    enabled = False
+
+from redash.query_runner import BaseQueryRunner, register
+from redash.query_runner import TYPE_STRING, TYPE_DATE, TYPE_DATETIME, TYPE_INTEGER, TYPE_FLOAT, TYPE_BOOLEAN
+from redash.utils import json_dumps, json_loads
+
+TYPES_MAP = {
+    0: TYPE_INTEGER,
+    1: TYPE_FLOAT,
+    2: TYPE_STRING,
+    3: TYPE_DATE,
+    4: TYPE_DATETIME,
+    5: TYPE_STRING,
+    6: TYPE_DATETIME,
+    13: TYPE_BOOLEAN
+}
+
+
+class Dgraph(BaseQueryRunner):
+    noop_query = """
+    {
+      test() {
+      }
+    }
+    """
+
+    @classmethod
+    def configuration_schema(cls):
+        return {
+            "type": "object",
+            "properties": {
+                "user": {
+                    "type": "string"
+                },
+                "password": {
+                    "type": "string"
+                },
+                "servers": {
+                    "type": "string"
+                }
+            },
+            "order": ["servers", "user", "password"],
+            "required": ["servers"],
+            "secret": ["password"]
+        }
+
+    @classmethod
+    def type(cls):
+        return "dgraph"
+
+    @classmethod
+    def enabled(cls):
+        return enabled
+
+    @classmethod
+    def remove_comments(cls, s):
+        """From https://stackoverflow.com/a/2319116
+
+        Remove all occurrences streamed comments (/*COMMENT */) from string"""
+        s = re.sub(re.compile("/\*.*?\*/", re.DOTALL), "", s)
+        return s
+
+    def run_query(self, query, user):
+
+        # gotta remove comments because Dgraph uses '#' as a comment delimiter, not '/* */'
+        query = Dgraph.remove_comments(query)
```

Comment:
You can remove the comment using `annotate_query`, like this:

https://github.com/getredash/redash/blob/261062d491876601f2153c8a14d1a42945d0b950/redash/query_runner/url.py#L7-L9